src/lib.rs

```diff
@@ -1,6 +1,29 @@
 use std::collections::HashMap;
 use std::collections::HashSet;
 use std::hash::Hash;
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct LengthError;
+
+
+impl fmt::Display for LengthError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "The lengths of the predicted and actual datasets must be equal.")
+    }
+}
+
+impl Error for LengthError {
+    fn description(&self) -> &str {
+        "The lengths of the predicted and actual datasets must be equal."
+    } 
+
+    fn cause(&self) -> Option<&Error> {
```

Comment:
I think there's a blanket implementation for `cause` already, so it can be omitted.

Reply:
Deleted 👍